src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/Huffman.cs

```diff
@@ -316,7 +578,8 @@ private static ushort[] GenerateDecodingLookupTree()
             // Create traverse path for all 0..256 octets, 256 is EOS, see: http://httpwg.org/specs/rfc7541.html#rfc.section.5.2
             for (int octet = 0; octet <= 256; octet++)
             {
-                (uint code, int bitLength) = Encode(octet);
+                uint code = s_encodingTableCodes[octet];
+                int bitLength = EncodingTableBitLengths[octet];
```

Comment:
Can you store both the array and the span into locals outside the loop, and then use those locals here?

Reply:
Done. 
Please is there some docs explaining best practice in this area? I was naively hoping compiler or JIT to optimize it.

Reply:
For the `ReadOnlySpan<byte>`, it's probably not necessary.  For the `uint[]`, it's likely necessary to get the best codegen until https://github.com/dotnet/runtime/pull/44562 lands.